### PR TITLE
chore: program anchor 0.30.1 compat

### PR DESCRIPTION
**Pull request by ananas-block**

### CODE REVIEW

File: programs/compressed-token/src/instructions/freeze.rs [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+
+use crate::program::LightCompressedToken;
 
 #[derive(Accounts)]
 pub struct FreezeInstruction<'info> {
     /// UNCHECKED: only pays fees.
     #[account(mut)]
     pub fee_payer: Signer<'info>,
-    #[account(address= mint.freeze_authority.unwrap()
+    #[account(constraint= authority.key() == mint.freeze_authority.unwrap()
```
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/instructions/freeze.rs:16]: what is the difference between these?

File: macros/light/src/accounts.rs [hunk 2]
```diff
@@ -10,12 +10,12 @@ pub(crate) fn process_light_accounts(input: DeriveInput) -> Result<TokenStream>
...
-                    "Program<'info, ::light_system_program::program::LightSystemProgram>",
+                    "Program<'info, ::LightSystemProgram>",
                 ),
                 ("system_program", "Program<'info, System>"),
                 (
                     "account_compression_program",
-                    "Program<'info, ::account_compression::program::AccountCompression>",
+                    "Program<'info, ::AccountCompression>",
                 ),
             ];
```
Comment (SwenSchaeferjohann) [inline macros/light/src/accounts.rs:20]: i believe this change causes issues when used by programs that don't import account_compression themselves. reverting the change should make token_escrow compile again IIRC

File: programs/compressed-token/src/instructions/freeze.rs (hunk 1 above)
Comment (ananas-block) [reply programs/compressed-token/src/instructions/freeze.rs:16]: address is not supported in anchor 0.30.1

**Review by SwenSchaeferjohann — APPROVED** [review]
lgmt. feel free to merge when green

**Pull request merged**